### ISSUE TITLE
feat: Add fused AdamW optimizer kernel

### ISSUE THREAD

**Issue by noahgift** [paiml/trueno#89]
## Summary

Add fused AdamW optimizer kernel to `trueno-gpu/src/kernels/optimizer/` for GPU-resident weight updates.

## Motivation

Currently, optimizer steps require:
1. Copy gradients GPU → CPU
2. Update weights on CPU
3. Copy weights CPU → GPU

This creates a severe bottleneck for training. A fused kernel eliminates all synchronization.

### Kernels to add:

1. **`AdamWStepKernel`** - Fused AdamW update:
```rust
/// θ = θ - lr * (m̂ / (√v̂ + ε) + λ * θ)
pub struct AdamWStepKernel { pub n: u32 }
```

2. **`AdamStepKernel`** - Adam without weight decay:
```rust
/// θ = θ - lr * m̂ / (√v̂ + ε)
pub struct AdamStepKernel { pub n: u32 }
```

3. **`GradientClipKernel`** - L2 gradient clipping:
```rust
/// g = g * min(1, max_norm / ‖g‖)
pub struct GradientClipKernel { pub n: u32 }
```

### Kernel parameters:
- `params`: weight tensor (in-place update)
- `grads`: gradient tensor
- `m`: first moment (updated in-place)
- `v`: second moment (updated in-place)
- `lr`, `beta1`, `beta2`, `eps`, `weight_decay`: hyperparameters
- `step`: current step for bias correction

## Acceptance Criteria

- [ ] `AdamWStepKernel` with fused bias correction + weight decay
- [ ] `AdamStepKernel` for vanilla Adam
- [ ] `GradientClipKernel` for L2 norm clipping
- [ ] Unit tests for PTX generation
- [ ] Exported from `kernels/mod.rs`

## References

- Phase 11, Week 4 of SPEC-FT-001
- entrenar optimizer integration
